src/Http/Routing/src/RouteValuesAddress.cs

```diff
@@ -24,4 +25,17 @@ public class RouteValuesAddress
     /// Gets or sets ambient route values from the current HTTP request.
     /// </summary>
     public RouteValueDictionary? AmbientValues { get; set; }
+
+    /// <inheritdoc />
+    public override string? ToString() => asString.Value;
+
+    readonly Lazy<string> asString;
+
+    /// <summary>
+    /// The constructor prepares the internal state of the object.
+    /// </summary>
+    public RouteValuesAddress()
+    {
+        asString = new Lazy<string>(() => $"{RouteName}({string.Join(',', ExplicitValues.Select(kv => $"{kv.Key}=[{kv.Value}]"))})");
+    }
```

Comment:
This doesn't need a public constructor. Caching this on the constructor is wrong since values can change after the first `ToString` call has been issued.

We also don't need to use `Lazy` as this code is not used in a multi-threaded context.

Reply:
Do you want to track changes to the values so that we could invalidate the cache?  This gets more and more Byzantine :scream: